src/serverModel.ts

```diff
@@ -222,19 +384,23 @@ export default function ServerModel(mpInstance) {
                 eventObject.LaunchReferral = window.location.href || null;
             }
 
+            // TODO: why is this duplicated with `uploadObject`?
             eventObject.CurrencyCode = mpInstance._Store.currencyCode;
             var currentUser = user || mpInstance.Identity.getCurrentUser();
-            self.appendUserInfo(currentUser, eventObject);
+            self.appendUserInfo(currentUser, eventObject as SDKEvent);
 
             if (event.messageType === Types.MessageType.SessionEnd) {
                 eventObject.SessionLength =
                     mpInstance._Store.dateLastEventSent.getTime() -
                     mpInstance._Store.sessionStartDate.getTime();
                 eventObject.currentSessionMPIDs =
                     mpInstance._Store.currentSessionMPIDs;
+
+                // TODO: Why are we using session attributes instead of the event's Event Attributes?
```

Comment:
Will have to experiment.  This might be a bug

Reply:
Ah. this is because it is inside of a session end event.  For a session end event, you cannot set an event attribute, it is simply:

mParticle.endSession().  

So we take the session attributes and turn them into event attributes for this.  Based on my quick search of the code, when you set a session attribute, it isn't actually ever sent with anything, and only with the session end event.